### PR TITLE
Add GitHub actions

### PR DESCRIPTION
**Pull request by fdiblen**
- removed travis configuration
- added GitHub actions
- added badges

Refs: #436

### CODE REVIEW

**Comment by sverhoeven** [comment]
Please undo the toc regenerate in README, this level of toc is unwanted

File: README.md [hunk 1]
```diff
@@ -1,30 +1,54 @@
...
-        - [Verifying the local installation](#verifying-the-local-installation)
-        - [Removing local state](#removing-local-state)
-    - [Customize your instance of the Research Software Directory](#customize-your-instance-of-the-research-software-directory)
-        - [General workflow when making changes](#general-workflow-when-making-changes)
-    - [Make your instance available to others by hosting it online (deployment)](#make-your-instance-available-to-others-by-hosting-it-online-deployment)
-        - [Configuring EC2](#configuring-ec2)
-        - [Configuring IAM](#configuring-iam)
-        - [Configuring S3](#configuring-s3)
-    - [Notes on security](#notes-on-security)
+  - [Try it out locally](#try-it-out-locally)
```
Comment (jspaaks) [inline README.md:17]: Can you restore this part like it was before?

File: .github/workflows/admin_tests.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  tests:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-latest]
+        node-version: [12.x, 14.x]
```
Comment (jspaaks) [inline .github/workflows/admin_tests.yml:16]: What's the idea behind doing 2 versions here?

File: .github/workflows/frontend_tests.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  tests:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-latest]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (jspaaks) [inline .github/workflows/frontend_tests.yml:16]: What's the idea behind 3 versions here? Shouldn't it (only) be the version used here https://github.com/research-software-directory/research-software-directory/blob/f5861de2c4b5373300110fc0dc08febb505a9709/frontend/Dockerfile#L1 ?

File: .github/workflows/frontend_tests.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        cd frontend
+        python -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (jspaaks) [inline .github/workflows/frontend_tests.yml:28]: Why the `if` clause? We control whether `requirements.txt` is there or not, right?

File: .github/workflows/admin_tests.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        node-version: ${{ matrix.node-version }}
+    - name: Cache Node.js modules
+      uses: actions/cache@v2
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.OS }}-node-
+          ${{ runner.OS }}-
```
Comment (jspaaks) [inline .github/workflows/admin_tests.yml:32]: I think it's cleaner to not reuse cache in testing

File: README.md (hunk 1 above)
Comment (fdiblen) [reply README.md]: it was auto generated. I restored it.

File: .github/workflows/backend_tests.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  tests:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-latest]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (jspaaks) [inline .github/workflows/backend_tests.yml:16]: Why 3 versions (see Frontend)

File: .github/workflows/admin_tests.yml (hunk 2 above)
Comment (fdiblen) [reply .github/workflows/admin_tests.yml:16]: 1. Node version used for development
2. the latest version of Node

File: .github/workflows/backend_tests.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        mongodb-version: [4.2]
```
Comment (jspaaks) [inline .github/workflows/backend_tests.yml:17]: That is not what we're using though: 
https://github.com/research-software-directory/research-software-directory/blob/f5861de2c4b5373300110fc0dc08febb505a9709/database/Dockerfile#L1

File: .github/workflows/backend_tests.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Start MongoDB
+      uses: supercharge/mongodb-github-action@1.3.0
+      with:
+        mongodb-version: ${{ matrix.mongodb-version }}
+    - name: Install dependencies
+      run: |
+        cd backend
+        python -m pip install --upgrade pip
+        pip install pytest==4.* pytest-cov coveralls
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (jspaaks) [inline .github/workflows/backend_tests.yml:34]: Why the if clause (see below)

File: .github/workflows/admin_tests.yml (hunk 2 above)
Comment (jspaaks) [reply .github/workflows/admin_tests.yml:16]: but how is 2) relevant here? Since we're not using it

**Review by jspaaks — CHANGES_REQUESTED** [review]
Really nice addition, thanks Faruk.

Left a couple of comments, none of which need to block merging of this PR excpet the TOC in the README; the others should be made into issues (some are already there, e.g. https://github.com/research-software-directory/research-software-directory/issues/425)

File: .github/workflows/admin_tests.yml (hunk 2 above)
Comment (fdiblen) [reply .github/workflows/admin_tests.yml:16]: using only 12.x now

File: .github/workflows/frontend_tests.yml (hunk 3 above)
Comment (fdiblen) [reply .github/workflows/frontend_tests.yml:16]: using only Python 3.6

File: .github/workflows/frontend_tests.yml (hunk 4 above)
Comment (fdiblen) [reply .github/workflows/frontend_tests.yml:28]: removed the if condition

File: .github/workflows/admin_tests.yml (hunk 5 above)
Comment (fdiblen) [reply .github/workflows/admin_tests.yml:32]: removed caching step

File: .github/workflows/backend_tests.yml (hunk 6 above)
Comment (fdiblen) [reply .github/workflows/backend_tests.yml:16]: using only Python 3.6

File: .github/workflows/backend_tests.yml (hunk 7 above)
Comment (fdiblen) [reply .github/workflows/backend_tests.yml:17]: Using only Mongodb 3.6

File: .github/workflows/backend_tests.yml (hunk 8 above)
Comment (fdiblen) [reply .github/workflows/backend_tests.yml]: removed the if condition

**Review by jspaaks — APPROVED** [review]
Nice job Faruk! :jack_o_lantern:

**Pull request merged**